packages/fxa-settings/src/pages/Signin/SigninUnblock/index.tsx

```diff
@@ -0,0 +1,212 @@
+  const [codeErrorMessage, setCodeErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [resendStatus, setResendStatus] = useState<ResendStatus>(
+    ResendStatus['not sent']
+  );
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const formAttributes: FormAttributes = {
+    inputFtlId: 'signin-unblock-code-input',
+    inputLabelText: 'Enter authorization code',
+    inputMode: InputModeEnum.text,
+    pattern: '[a-zA-Z0-9]{8}',
+    maxLength: 8,
+    submitButtonFtlId: 'signin-unblock-submit-button',
+    submitButtonText: 'Continue',
+  };
+
+  const localizedCustomCodeRequiredMessage = ftlMsgResolver.getMsg(
+    'signin-unblock-code-required-error',
+    'Authorization code required'
+  );
+
+  const isValidCodeFormat = (unblockCode: string) => {
+    return /^[a-zA-Z0-9]+$/.test(unblockCode);
+  };
+
+  const isValidLength = (unblockCode: string) => {
+    return unblockCode.length === 8;
+  };
+
+  const onSubmit = async (unblockCode: string) => {
+    setBannerErrorMessage('');
+    setResendStatus(ResendStatus['not sent']);
+    if (!isValidLength(unblockCode)) {
+      setCodeErrorMessage(
+        ftlMsgResolver.getMsg(
+          'signin-unblock-code-incorrect-length',
+          'Authorization code must contain 8 characters'
+        )
+      );
+    } else if (!isValidCodeFormat(unblockCode)) {
+      setCodeErrorMessage(
+        ftlMsgResolver.getMsg(
+          'signin-unblock-code-incorrect-format',
+          'Authorization can only contain letters and/or numbers'
+        )
+      );
+    } else {
+      setIsLoading(true);
+      GleanMetrics.login.submit();
+      const { data, error } = await signinWithUnblockCode(unblockCode);
+      if (data) {
+        GleanMetrics.login.success();
+
+        const accountData: StoredAccountData = {
+          email,
+          uid: data.signIn.uid,
+          lastLogin: Date.now(),
+          sessionToken: data.signIn.sessionToken,
+          verified: data.signIn.verified,
+          metricsEnabled: data.signIn.metricsEnabled,
+        };
+
+        storeAccountData(accountData);
+
+        const navigationOptions: NavigationOptions = {
+          verified: data.signIn.verified,
+          verificationMethod: data.signIn.verificationMethod,
+          verificationReason: data.signIn.verificationReason,
+          wantsTwoStepAuthentication,
+        };
+
+        const navigationArgs = getNavigationTarget(navigationOptions);
+        navigate(navigationArgs.to, navigationArgs.options);
+      }
+      if (error) {
+        const localizedErrorMessage = getLocalizedErrorMessage(
+          ftlMsgResolver,
+          error
+        );
+        if (error.errno === AuthUiErrors.INCORRECT_PASSWORD.errno) {
+          navigate(`/signin${location.search}`, {
+            state: {
+              email,
+              hasLinkedAccount,
+              hasPassword,
```

Comment:
Oof, it's painful we have to go full circle passing these all around... Looks like we're receiving `hasLinkedAccount` and `hasPassword` from `signin`, taking users here, then passing it back.

It would be nice to do this here, but I think this is also OK as a follow up (with comments in code, maybe we can update FXA-9177?) - whenever we fetch or receive these properties in the Signin container, we should just update the apollo cache value so we can later query `account.linkedAccounts` and whatever we need for `hasPassword`. You can see where I did something similar in the `cachedSigninHandler` in the Signin container.

Reply:
Added TODOs in code and to 9177 ticket

Reply:
Sweet, thanks!